### PR TITLE
GEOMESA-2165 Updates to tutorials documentation

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: docs/tutorials/geomesa-quickstart-lambda.rst [hunk 1]
```diff
@@ -125,17 +231,35 @@ start finishes, all the feature should be returned from the persistent store (Ac
 Looking at the Code
 -------------------
 
+The source code is meant to be accessible for this tutorial. The logic is contained in
+the generic ``org.geomesa.example.quickstart.GeoMesaQuickStart`` in the ``geomesa-quickstart-common`` module,
+and the Kafka-specific ``org.geomesa.example.lambda.LambdaQuickStart`` in the ``geomesa-quickstart-lambda`` module.
```
Comment (aheyne) [inline docs/tutorials/geomesa-quickstart-lambda.rst]: Do we want to say `lambda-specific` or `kafka/accumulo specific` or something here?

File: docs/tutorials/geomesa-quickstart-hbase.rst [hunk 2]
```diff
@@ -5,266 +5,281 @@ This tutorial is the fastest and easiest way to get started with the
...
+    07 719025743=719025743|AUTHORITIES||||051|60|12|60|3|Wichita, Kansas, United States|US|2018-01-01T00:00:00.000Z|POINT (-97.3375 37.6922)
+    08 719027205=719027205|UNITED STATES|USA|SIOUX||051|4|1|4|3|Sioux City, Iowa, United States|US|2018-01-01T00:00:00.000Z|POINT (-96.4003 42.5)
+    09 719025111=719025111|||UNITED STATES|USA|051|2|1|2|3|Pickens County, South Carolina, United States|US|2018-01-01T00:00:00.000Z|POINT (-82.7165 34.9168)
+    10 719026938=719026938|PITTSBURGH|USA|||051|5|1|5|3|York County, Pennsylvania, United States|US|2018-01-01T00:00:00.000Z|POINT (-77 40.1254)
+
+    Returned 138 total features
+
+    Running query EventCode = '051' AND dtg DURING 2017-12-31T00:00:00+00:00/2018-01-02T00:00:00+00:00
+    Returning attributes [GLOBALEVENTID, dtg, geom]
+    01 719024909=719024909|2018-01-01T00:00:00.000Z|POINT (144.967 -37.8167)
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-hbase.rst]: yeah it works, sphinx ignores replacements in code blocks

File: docs/tutorials/geomesa-quickstart-accumulo.rst [hunk 3]
```diff
@@ -103,97 +101,134 @@ where you provide the following arguments:
...
-Register the GeoMesa store with GeoServer
+The source code is meant to be accessible for this tutorial. The main logic is contained in
+the generic ``org.geomesa.example.quickstart.GeoMesaQuickStart`` in the ``geomesa-quickstart-common`` module,
+which is datastore agnostic. Some relevant methods are:
+
+-  ``createDataStore`` get a datastore instance from the input configuration
+-  ``createSchema`` create the schema in the datastore, as a pre-requisite to writing data
+-  ``writeFeatures`` use a ``FeatureWriter`` to write features to the datastore
+-  ``queryFeatures`` run several queries against the datastore
+-  ``cleanup`` deletes data and disposes of the datastore instance
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-accumulo.rst]: changed to "delete the sample data and dispose of the datastore instance"

File: docs/tutorials/geomesa-quickstart-cassandra.rst [hunk 4]
```diff
@@ -1,48 +1,40 @@
...
+You will need a namespace in Cassandra to contain the tutorial tables. The easiest way to do
+this is with the ``cqlsh`` tool provided with Cassandra distributions. Start ``cqlsh``, then type:
 
 .. code-block:: bash
 
-    cqlsh>  CREATE KEYSPACE mykeyspace WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor' : 3};
+    cqlsh>  CREATE KEYSPACE geomesa WITH REPLICATION = {'class': 'SimpleStrategy', 'replication_factor' : 3};
 
-This creates a key space called “mykeyspace”. This is a top-level name
+This creates a key space called "gemesa". This is a top-level name
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-cassandra.rst]: good catch

File: docs/tutorials/geomesa-quickstart-cassandra.rst [hunk 5]
```diff
@@ -57,200 +49,199 @@ Pick a reasonable directory on your machine, and run:
...
-profile:
+- ``<host:port>`` the hostname and port your Cassandra instance is
+  running on. For Cassandra standalone this will be ``localhost:9042``. More
+  information on how to find your connection is available
+  `here <http://www.geomesa.org/documentation/user/cassandra/install.html#connecting-to-cassandra>`__.
+- ``<keyspace>`` keyspace your table will be put into. If you followed the instructions above,
+  this will be ``geomesa``. More information on how to setup keyspaces is available
+  `here <http://www.geomesa.org/documentation/user/cassandra/install.html#connecting-to-cassandra>`__.
+- ``<table>`` the name of the destination table that will
+  accept these test records; this table should either not exist or
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-cassandra.rst]: changed to a `.`

File: docs/tutorials/geomesa-quickstart-accumulo.rst [hunk 6]
```diff
@@ -233,25 +261,12 @@ of the layer you just created into the search box, and press
...
-Once you press ``<Enter>``, the display will update, and you should see
-a collection of red dots similar to the following image.
+a new tab. You should see a collection of red dots similar to the following image:
 
 .. figure:: _static/geomesa-quickstart-accumulo/geoserver-layer-preview.png
-   :alt: Visualizing quick-start data
+    :alt: Visualizing quick-start data
 
-   Visualizing quick-start data
+    Visualizing quick-start data
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-accumulo.rst]: hmm, suggestions?

File: docs/tutorials/geomesa-quickstart-hbase.rst [hunk 7]
```diff
@@ -5,266 +5,281 @@ This tutorial is the fastest and easiest way to get started with the
...
 
--  `Java <http://java.oracle.com/>`__ Development Kit 1.8,
--  Apache `Maven <http://maven.apache.org/>`__,
--  a GitHub client,
--  HBase 1.2.x (optional), and
--  GeoServer |geoserver_version| (optional).
+-  `Java <http://java.oracle.com/>`__ JDK 1.8
+-  Apache `Maven <http://maven.apache.org/>`__ |maven_version|
+-  a GitHub client
+-  an HBase 1.3.x instance
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-hbase.rst]: added

File: docs/tutorials/geomesa-quickstart-hbase.rst [hunk 8]
```diff
@@ -5,266 +5,281 @@ This tutorial is the fastest and easiest way to get started with the
...
-link.
-
-You will be taken to the "Edit Layer" screen.
+GeoServer should recognize the ``gdelt-quickstart`` feature type, and
+should present that as a layer that can be published. Click on the
+"Publish" link.
 
-In the "Data" pane, enter values for the bounding boxes. In this case,
-you can click on the link to compute these values from the data.
+You will be taken to the Edit Layer screen. You will need to enter values for the data bounding
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-hbase.rst]: quoted

File: docs/tutorials/geomesa-quickstart-accumulo.rst [hunk 9]
```diff
@@ -41,32 +46,25 @@ Pick a reasonable directory on your machine, and run:
...
-
-.. code-block:: bash
+.. warning::
 
-    $ mvn clean install -pl geomesa-quickstart-accumulo
+    Make sure that you download or checkout the version of the tutorials project that corresponds to
+    your GeoMesa version. See :ref:`tutorial_versions` for more details.
 
-.. note::
+To ensure that the quick start works with your environment, modify the ``pom.xml``
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-accumulo.rst]: i think we use `quick start` in most cases, and `quick-start` when used as an adjective, which I believe is correct. i just did a quick check and that seemed to be consistent. do you see any specific ones that are wrong?

File: docs/tutorials/geomesa-quickstart-cassandra.rst [hunk 10]
```diff
@@ -57,200 +49,199 @@ Pick a reasonable directory on your machine, and run:
...
 After the insertions are complete, a sequence of queries are run to
 demonstrate different types of queries possible via the GeoTools API.
 
-Run the Tutorial
-----------------
+Running the Tutorial
+--------------------
 
-On the command-line, run:
+On the command line, run:
```
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-cassandra.rst]: fixed to be `command line` except when used as an adjective.

File: docs/tutorials/geomesa-quickstart-lambda.rst (hunk 1 above)
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-lambda.rst]: changed to "Kafka/Accumulo-specific"

File: docs/tutorials/geomesa-quickstart-accumulo.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (aheyne) [reply docs/tutorials/geomesa-quickstart-accumulo.rst]: I guess this is here to just be sub text under an image?

File: docs/tutorials/geomesa-quickstart-accumulo.rst (hunk 9 above)
[the remark this answers is not in the document]
Comment (aheyne) [reply docs/tutorials/geomesa-quickstart-accumulo.rst]: Nope, just ran through them all and they all look consistent with this logic.

File: docs/tutorials/geomesa-quickstart-accumulo.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (elahrvivaz) [reply docs/tutorials/geomesa-quickstart-accumulo.rst]: oh, i thought you mean the whole section was out of place - yeah, there is an 'alt' that shows up if you hover, and this is the visible caption for the image.

**Comment by elahrvivaz** [comment]
Pushed more updates to correspond with https://github.com/geomesa/geomesa-tutorials/pull/60

**Pull request merged**